File: app/src/main/java/app/xunxun/homeclock/preferences/BackgroundModePreferencesDao.java
package app.xunxun.homeclock.preferences;

import android.content.Context;
import android.content.SharedPreferences;
import android.preference.PreferenceManager;

/**
 * Created by fengdianxun on 15-6-24.
 */
public class BackgroundModePreferencesDao {
    private static final String KEY = "BackgroundModePreferencesDao";
    public static final int MODE_COLOR = 0;
    public static final int MODE_ONLINE_IMAGE = 1;
    public static final int MODE_LOCAL_IMAGE = 2;

    public static void set(Context context, int flag) {
        SharedPreferences prefs = PreferenceManager.getDefaultSharedPreferences(context);
        SharedPreferences.Editor editor = prefs.edit();
        editor.putInt(KEY, flag);
        editor.apply();
    }

    public static int get(Context context) {
        SharedPreferences prefs = PreferenceManager.getDefaultSharedPreferences(context);
        return prefs.getInt(KEY, 0);
    }

}

File: app/src/main/java/app/xunxun/homeclock/preferences/BackgroundModePreferencesDao.kt
package app.xunxun.homeclock.preferences

import android.content.Context
import android.content.SharedPreferences
import android.preference.PreferenceManager

/**
 * Created by fengdianxun on 15-6-24.
 */
object BackgroundModePreferencesDao {
    private val KEY = "BackgroundModePreferencesDao"
    val MODE_COLOR = 0
    val MODE_ONLINE_IMAGE = 1
    val MODE_LOCAL_IMAGE = 2

    operator fun set(context: Context, flag: Int) {
        val prefs = PreferenceManager.getDefaultSharedPreferences(context)
        val editor = prefs.edit()
        editor.putInt(KEY, flag)
        editor.apply()
    }

    operator fun get(context: Context): Int {
        val prefs = PreferenceManager.getDefaultSharedPreferences(context)
        return prefs.getInt(KEY, 0)
    }

}
